sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/education-program-offering.institutions.createOffering.e2e-spec.ts

```diff
@@ -0,0 +1,161 @@
+      fakeEducationProgram,
+    );
+    const endpoint = `/institutions/education-program-offering/location/${collegeFLocation.id}/education-program/${savedFakeEducationProgram.id}`;
+    const payload = {
+      offeringName: "Offering 1",
+      yearOfStudy: 1,
+      showYearOfStudy: true,
+      offeringIntensity: OfferingIntensity.fullTime,
+      offeringDelivered: OfferingDeliveryOptions.Onsite,
+      hasOfferingWILComponent: "no",
+      studyStartDate: "2023-09-01",
+      studyEndDate: "2024-06-30",
+      lacksStudyBreaks: false,
+      studyBreaks: [
+        {
+          breakStartDate: "2023-12-01",
+          breakEndDate: "2024-01-01",
+          breakDays: 32,
+          eligibleBreakDays: 21,
+          ineligibleBreakDays: 11,
+        },
+      ],
+      studyBreaksOverlappingDetection: false,
+      studyPeriodBreakdown: {
+        totalDays: 304,
+        totalFundedWeeks: 42,
+        fundedStudyPeriodDays: 293,
+        unfundedStudyPeriodDays: 11,
+      },
+      offeringType: OfferingTypes.Public,
+      offeringDeclaration: true,
+      mode: "editable",
+      validationWarnings: [],
+      validationInfos: ["invalidStudyBreakAmountOfDays"],
+      applicationId: "",
+      applicationStatus: "",
+      actualTuitionCosts: 1234,
+      programRelatedCosts: 3211,
+      mandatoryFees: 456,
+      exceptionalExpenses: 555,
+    };
+
+    // Act/Assert
+    let educationProgramOfferingId: number;
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+        educationProgramOfferingId = response.body.id;
+      });
+    const createdEducationProgramOffering =
+      await db.educationProgramOffering.findOne({
+        where: { id: educationProgramOfferingId },
+      });
+    expect(createdEducationProgramOffering).toEqual(
+      expect.objectContaining({
+        name: payload.offeringName,
+        studyStartDate: payload.studyStartDate,
+        studyEndDate: payload.studyEndDate,
+        actualTuitionCosts: payload.actualTuitionCosts,
+        programRelatedCosts: payload.programRelatedCosts,
+        mandatoryFees: payload.mandatoryFees,
+        exceptionalExpenses: payload.exceptionalExpenses,
+        offeringDelivered: payload.offeringDelivered,
+        lacksStudyBreaks: payload.lacksStudyBreaks,
+        offeringType: payload.offeringType,
+        offeringIntensity: payload.offeringIntensity,
+        yearOfStudy: payload.yearOfStudy,
+        showYearOfStudy: payload.showYearOfStudy,
+        hasOfferingWILComponent: payload.hasOfferingWILComponent,
+        offeringWILType: null,
+        studyBreaks: {
+          totalDays: payload.studyPeriodBreakdown.totalDays,
+          studyBreaks: [
+            {
+              breakDays: payload.studyBreaks[0].breakDays,
```

Comment:
Usually, we try to avoid accessing the same index multiple times to improve the readability, for instance, by creating a variable to store the `payload.studyBreaks[0]`.